src/test_l3_dsFPD.c

```diff
@@ -0,0 +1,627 @@
+    assert(status == dsERR_NONE);
+    /* Check that the Indicator is valid */
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to set Front panel LED Blink functionality.
+*
+* This test case provides a scope to  set the Blink duration and iteration 
+* of FP indicator.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+* Front panel idicator state should be set to On through Test 2 before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 4 and provide indicator type, Blink duration
+* and Blink iteration.
+*
+*/
+void test_4_dsFPD_hal_SetFPBlink(void)
+{
+    gTestID = 4;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    uint32_t uBlinkDuration;
+    uint32_t uBlinkIterations;
+
+    selectIndicator();
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Enter Blink Duration in ms: ");
+    scanf("%d", &uBlinkDuration);
+    readAndDiscardRestOfLine(stdin);
+
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Enter Blink iteration: ");
+    scanf("%d", &uBlinkIterations);
+    readAndDiscardRestOfLine(stdin);
+
+
+    /* Check that the Indicator is valid */
+    UT_LOG_INFO("Calling dsSetFPBlink(IN:Indicator:[0x%d], IN:Blink Duration in ms:[%d], IN:Blink Iteration:[%d])" \
+                                        ,eIndicator,uBlinkDuration,uBlinkIterations);
+    status = dsSetFPBlink((dsFPDIndicator_t)eIndicator,uBlinkDuration,uBlinkIterations);
+    UT_LOG_INFO("Result dsSetFPBlink()dsError_t:[%s]",UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+    assert(status == dsERR_NONE);
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to set Front panel LED Brightness functionality.
+*
+* This test case provides a scope to  set the Brightness of selected FP indicator.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+* Front panel indicator state should be set to ON through Test 2 before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 5 and provide indicator index and Brightness value.
+*
+*/
+void test_5_dsFPD_hal_SetFPBrightness(void)
+{
+    gTestID = 5;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    uint32_t uBrightness;
+
+    selectIndicator();
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Enter LED Brightness (Range: 0-100): ");
+    scanf("%d", &uBrightness);
```

Comment:
validate the user-entered values if they meet the range.  All through the file.

Reply:
updated